src/mps/dmrg.jl

```diff
@@ -278,7 +278,15 @@ function dmrg(PH, psi0::MPS, sweeps::Sweeps; kwargs...)
         end
 
         energy = vals[1]
-        phi::ITensor = vecs[1]
+        ## Right now there is a conversion problem in CUDA.jl where `UnifiedMemory` Arrays are being converted 
+        ## into `DeviceMemory`. This conversion line is here temporarily to fix that problem when it arises
+        ## Adapt is only called when using CUDA backend. CPU will work as implemented previously.
+        phi::ITensor = if NDTensors.iscu(tensor(phi))
```

Comment:
Let's define `iscu(A::ITensor) = iscu(tensor(A))` somewhere since I'm sure this is something we will want to use in other places.

Also I think we should also check `iscu(vecs[1])` as well since this may accidentally move `vecs[1]` to GPU in cases where it was supposed to be on CPU.